bridge_ui/src/components/Transactions.tsx

```diff
@@ -0,0 +1,519 @@
+    console.log("failed to update txs status, error: " + error)
+  })
+}
+
+function ListTransaction({status, sourceChainId, targetChainId}: {
+  status: TxStatus,
+  sourceChainId: ChainId,
+  targetChainId: ChainId
+}) {
+  const classes = useStyles()
+  const [open, setOpen] = useState(false)
+
+  const handleClick = () => {
+    setOpen(!open)
+  }
+  const transactions = useLiveQuery(
+    async () => await getTxsByStatus(status, sourceChainId, targetChainId),
+    [status, sourceChainId]
+  )
+
+  return (
+    <List
+      component="nav"
+      className={classes.list}
+    >
+      <ListItem button onClick={handleClick}>
+        <ListItemText primary={status + " Transactions"} />
+        {open ? <ExpandLess /> : <ExpandMore />}
+      </ListItem>
+      <Collapse in={open} timeout="auto" unmountOnExit>
+        <List
+          component="div" disablePadding
+          style={{maxHeight:200, overflow: 'auto'}}
+        >
+          {transactions?.map((tx: Transaction) => {
+            return (
+              <ListItem className={classes.nested} key={tx.txId}>
+                <Typography component='div'>
+                  <Box fontFamily="Monospace" fontWeight="fontWeightMedium">{tx.txId}</Box>
+                </Typography>
+              </ListItem>
+            )
+          })}
+        </List>
+      </Collapse>
+    </List>
+  );
+}
+
+// only show transactions from Alephium to ETH
+export default function Transactions() {
+  const classes = useStyles()
+  const [type, setType] = useState<"Token" | "NFT">("Token")
+  const isNFT = type === "NFT";
+  const [txSourceChain, setTxSourceChain] =
+    useState<ChainId>(CHAIN_ID_ALEPHIUM);
+
+  const [txTargetChain, setTxTargetChain] =
+    useState<ChainId>(CHAIN_ID_ETH)
+
+  const [pageNumber, setPageNumber] = useState<number>(1)
+  const [syncInfo, setSyncInfo] = useState<SyncInfo | undefined>(undefined)
+  const [resync, setResync] = useState<boolean>(false)
+
+  const { isReady: sourceChainReady } = useIsWalletReady(txSourceChain)
+  const { isReady: targetChainReady } = useIsWalletReady(txTargetChain)
+
+  const { signer: alphSigner } = useAlephiumWallet()
+  const { provider: ethProvider } = useEthereumProvider()
+  const [isLoading, setIsLoading] = useState(true)
+  const [loadingError, setLoadingError] = useState<any>(null)
+
+  const isTransferCompleted = useCallback(async (sequence: string) => {
+    if (ethProvider && txTargetChain === CHAIN_ID_ETH) {
+      const response = await getSignedVAAWithRetry(
+        CHAIN_ID_ALEPHIUM,
+        ALEPHIUM_TOKEN_BRIDGE_CONTRACT_ID,
+        CHAIN_ID_ETH,
+        sequence
+      )
+      return getIsTransferCompletedEth(
+        getTokenBridgeAddressForChain(CHAIN_ID_ETH),
+        ethProvider,
+        response.vaaBytes
+      )
+    }
+    throw Error("Only support ETH as target chain")
+  }, [ethProvider, txTargetChain])
+
+  // load more transactions from explorer
+  const loadMoreTxsCallback = useCallback(async () => {
```

Comment:
I will think how to make this more efficient.

Does Wormhole's UI uses Etherscan for Tx history?

Reply:
No, the wormhole bridge UI won't support tx history, users need to search from Etherscan by themself.